### PR TITLE
Fail when regression image is missing on CI

### LINKED ISSUE #2766: Add fail option if image is not exist to `verify_cache_image` for CI
### Describe what maintenance you would like added.

Add an option to `verify_cache_image` that fails if the image does not exist in the `image_cache` directory.

_Originally posted by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/2765#issuecomment-1152798367_

### Links to source code.

https://github.com/pyvista/pyvista/blob/fc9cea791d3d8aa873ba3123ee9d1c0e8d065faa/tests/plotting/test_plotting.py#L161-L167

### Pseudocode or Screenshots

```python
@skip_no_plotting
def test_add_background_image_not_global():
    plotter = pyvista.Plotter()
    plotter.add_mesh(sphere)
    plotter.show(before_close_callback=verify_cache_image)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview

 

Closes #2766.  This PR adds option to pytest to fail image regression test when image is missing.

This is used is in the CI.

### Details

The first commit has a test without an image to test CI, will make "ready to review" when removed.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -101,3 +101,9 @@ def pointset():
     rng = default_rng(0)
     points = rng.random((10, 3))
     return pyvista.PointSet(points)
+
+
+def pytest_addoption(parser):
```
Comment (MatthewFlamm) [inline tests/conftest.py:106]: Needed to move these up a level.  Previously these would only be set when calling `pytest tests/plotting ...` but but when ``pytest tests ...`.  I do not see a downside to allowing this more flexible method for all options.

**Comment by MatthewFlamm** [comment]
The test with no image failed here:

https://github.com/pyvista/pyvista/runs/6851480441?check_suite_focus=true#step:10:1480

with:

```
RuntimeError: /home/runner/work/pyvista/pyvista/tests/plotting/image_cache/no_image.png does not exist in image cache
```

File: tests/conftest.py (hunk 1 above)
Comment (akaszynski) [reply tests/conftest.py:106]: Agreed.

**Review by akaszynski — APPROVED** [review]
LGTM

**Review by tkoyama010 — APPROVED** [review]
Awesome job. I didn't notice idea to use option.

**Pull request merged**